packages/client/src/pages/CheckerPage.tsx

```diff
@@ -1,313 +1,336 @@
-    // In case no active sections are found, reset the checker
-    // This is a fallback to prevent users being stuck without any active component
-    const activeComponent = [
-      sections.OUTCOME,
-      sections.LOCATION_INPUT,
-      sections.QUESTIONS,
-    ].find((section) => isActive(section));
-
-    if (!activeComponent) {
-      console.error("Resetting checker, because no active section was found");
-      setTopicData(defaultTopicSession);
-    }
-
-    // Prevent linter to add all dependencies, now the useEffect is only called on mount
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    // Count the times the checker is loaded
+    setTopicData({
+      timesCheckerLoaded: timesCheckerLoaded + 1,
+    });
+    // eslint-disable-next-line
   }, []);
 
-  const isActive = useCallback(
-    (component: string[] | string, finished: boolean = false) => {
-      // If component is only a string, we make it a array first
-      const allComponents = Array.isArray(component) ? component : [component];
-
-      // If finished is true we check if it's finished, else check activeComponents.
-      const components = finished
-        ? finishedComponents
-        : activeComponents
-        ? activeComponents
-        : [];
-      return components.includes(allComponents[0]);
-    },
-    [activeComponents, finishedComponents]
-  );
-
-  // Only one component can be active at the same time.
-  const setActiveState = (component: string) => {
-    // Do not track the active step
-    if (!isActive(component)) {
-      matomoTrackEvent({
-        action: actions.ACTIVE_STEP,
-        name: component,
+  useEffect(() => {
+    if (checker) {
+      // Potentially restore new sections from session
+      const restoreTopicData =
+        sectionData.length === defaultSections.length &&
+        sectionData.filter((s) => s.isActive).length === 1;
+
+      // Initialize new sections
+      defaultSections.map((section, index) => {
+        // Reorder the indexes (in case sections have been added)
+        section.index = index;
+
+        if (restoreTopicData) {
+          // Restore sectionData from session
+          section.isActive = sectionData[index].isActive;
+          section.isCompleted = sectionData[index].isCompleted;
+        } else {
+          // Set the first section active
+          section.isActive = index === 0;
+          // Set all sections incomplete
+          section.isCompleted = false;
+        }
+
+        return section;
       });
-    }
 
-    setTopicData({ activeComponents: [component] });
-  };
+      // Autocomplete the location section in case a new checker with data needs was started with a stored address
+      if (address && hasDataNeeds && !sectionData.length) {
+        // Complete the location section
+        defaultSections[0].isActive = false;
+        defaultSections[0].isCompleted = true;
+        // Activate the question section
+        defaultSections[1].isActive = true;
```

Comment:
Right now the situation is still specific for our use case. 

How does this work on default? 
When only specifying sections and start at 0

Reply:
Good point. Fixed this in [this commit](https://github.com/Amsterdam/vergunningcheck/pull/758/commits/b9fd958e28df87022ef2bcbcde2f8104b5a1bc29)